airflow/contrib/operators/bigquery_operator.py

```diff
@@ -192,24 +192,46 @@ def execute(self, context):
             )
             conn = hook.get_conn()
             self.bq_cursor = conn.cursor()
-        job_id = self.bq_cursor.run_query(
-            sql=self.sql,
-            destination_dataset_table=self.destination_dataset_table,
-            write_disposition=self.write_disposition,
-            allow_large_results=self.allow_large_results,
-            flatten_results=self.flatten_results,
-            udf_config=self.udf_config,
-            maximum_billing_tier=self.maximum_billing_tier,
-            maximum_bytes_billed=self.maximum_bytes_billed,
-            create_disposition=self.create_disposition,
-            query_params=self.query_params,
-            labels=self.labels,
-            schema_update_options=self.schema_update_options,
-            priority=self.priority,
-            time_partitioning=self.time_partitioning,
-            api_resource_configs=self.api_resource_configs,
-            cluster_fields=self.cluster_fields,
-        )
+        if isinstance(self.sql, str):
+            job_id = self.bq_cursor.run_query(
+                sql=self.sql,
+                destination_dataset_table=self.destination_dataset_table,
+                write_disposition=self.write_disposition,
+                allow_large_results=self.allow_large_results,
+                flatten_results=self.flatten_results,
+                udf_config=self.udf_config,
+                maximum_billing_tier=self.maximum_billing_tier,
+                maximum_bytes_billed=self.maximum_bytes_billed,
+                create_disposition=self.create_disposition,
+                query_params=self.query_params,
+                labels=self.labels,
+                schema_update_options=self.schema_update_options,
+                priority=self.priority,
+                time_partitioning=self.time_partitioning,
+                api_resource_configs=self.api_resource_configs,
+                cluster_fields=self.cluster_fields,
+            )
+        else:
```

Comment:
Great caching this! 

One small thing - I think it would be nice for the user to check if self.sql is an iterable and throw a nice exception if it is neither str nor iterable  - note that sequence matters, str is also an iterable. Alternatively you could introduce Python3 type annotations (which we are adding gradually).

Reply:
Good point. I was thinking about that non-iterable case. The question is, do we allow dicts, which are iterables as well?

Regarding annotations, could we have mixed type annotations, like 
```def fun(sql: str or iterable):```
 or should we just annotate with `iterable`?

Reply:
added an additional check and type annotations.